src/main/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateAddColumnsBigQueryStep.java

```diff
@@ -0,0 +1,96 @@
+package bio.terra.service.dataset.flight.update;
+
+import bio.terra.common.Column;
+import bio.terra.model.ColumnModel;
+import bio.terra.model.DatasetSchemaUpdateModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetJsonConversion;
+import bio.terra.service.dataset.DatasetTable;
+import bio.terra.service.tabulardata.google.BigQueryProject;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import com.google.cloud.bigquery.BigQuery;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class DatasetSchemaUpdateAddColumnsBigQueryStep implements Step {
+  private final BigQueryDatasetPdao bigQueryDatasetPdao;
+  private final DatasetDao datasetDao;
+  private final UUID datasetId;
+  private final DatasetSchemaUpdateModel updateModel;
+
+  private static final List<String> EMPTY_LIST = Collections.emptyList();
```

Comment:
Instead of `Collections.emptyList()`, the form `List.of()` is better as it matches the API used to create lists. In either case, there's no need to create your own constant, as the result of both of those functions are already constant.